src/main.rs

```diff
@@ -116,9 +117,15 @@ fn packages_from_git(repo: &Repository, hash: Oid, path: &Path) -> Result<Deps,
     Ok(deps)
 }
 
-fn get_changelog(path: &Path) -> String {
+fn get_changelog(path: &Path) -> Result<String, Error> {
     let changelog_file = path.join("CHANGELOG.md");
-    let contents = fs::read_to_string(changelog_file).unwrap_or("".to_string());
+    let contents = match fs::read_to_string(changelog_file) {
+        Ok(ok) => Ok(ok),
+        Err(err) => match err.kind() {
+            ErrorKind::NotFound => Ok(String::new()),
+            _ => Err(anyhow!("Error while reading CHANGELOG.md")),
```

Comment:
Why do you throw the specific error out of the window here? :confused: Why not something like `Err(Error::from(err).context(format!("Error while reading {}", changelog_file.display)))` to preserve it including the actual path? (That'll require to pass `&changelog_file` to `read_to_string`).

Reply:
Thank you! That's what I needed :pray: